diff --git a/wireproxy/wireguard.go b/wireproxy/wireguard.go
index 661504b..0165d96 100644
--- a/wireproxy/wireguard.go
+++ b/wireproxy/wireguard.go
@@ -3,6 +3,7 @@ package wireproxy
 import (
 	"bytes"
 	"fmt"
+	"log"
 
 	"net/netip"
 
@@ -63,7 +64,10 @@ func StartWireguard(conf *DeviceConfig) (*VirtualTun, error) {
 	if err != nil {
 		return nil, err
 	}
-	dev := device.NewDevice(tun, conn.NewDefaultBind(), device.NewLogger(device.LogLevelVerbose, ""))
+	dev := device.NewDevice(tun, conn.NewDefaultBind(), &device.Logger{
+		Verbosef: func(f string, v ...any) { log.Output(2, "[wg] [D] "+fmt.Sprintf(f, v...)) },
+		Errorf:   func(f string, v ...any) { log.Output(2, "[wg] [E] "+fmt.Sprintf(f, v...)) },
+	})
 	err = dev.IpcSet(setting.ipcRequest)
 	if err != nil {
 		return nil, err
