diff --git a/extras/wireproxy/wireguard.go b/extras/wireproxy/wireguard.go
index b98bc35..8e219e9 100644
--- a/extras/wireproxy/wireguard.go
+++ b/extras/wireproxy/wireguard.go
@@ -3,6 +3,7 @@ package wireproxy
 import (
 	"bytes"
 	"fmt"
+	"log"
 
 	"net/netip"
 
@@ -69,7 +70,16 @@ func StartWireguard(conf *DeviceConfig, logLevel int) (*VirtualTun, error) {
 	if err != nil {
 		return nil, err
 	}
-	dev := device.NewDevice(tun, conn.NewDefaultBind(), device.NewLogger(logLevel, ""))
+
+	logger := &device.Logger{Verbosef: device.DiscardLogf, Errorf: device.DiscardLogf}
+	if logLevel >= device.LogLevelVerbose {
+		logger.Verbosef = func(f string, v ...any) { log.Output(2, "[wg] [D] "+fmt.Sprintf(f, v...)) }
+	}
+	if logLevel >= device.LogLevelError {
+		logger.Errorf = func(f string, v ...any) { log.Output(2, "[wg] [E] "+fmt.Sprintf(f, v...)) }
+	}
+
+	dev := device.NewDevice(tun, conn.NewDefaultBind(), logger)
 	err = dev.IpcSet(setting.ipcRequest)
 	if err != nil {
 		return nil, err
